src/styles/theme.md

```diff
@@ -0,0 +1,155 @@
+  document.getElementById("root")
+);
+```
+
+## Accesing theme in a component
+
+If you wish to style material UI components from the theme, you can do so by using the `useTheme` hook. This hook gives you access to the application's theme and allows you to use the theme variables inside your components.
+
+```jsx
+import { useTheme } from "@material-ui/core/styles";
+
+function componentUsingTheme() {
+  const theme = useTheme();
+  return <span>{`spacing ${theme.spacing}`}</span>;
+}
+```
+
+## How to change color palette
+
+The `Palette` object is one of the important variables to configure, as it enables you to modify the color of components to suit your needs. In the example below, blue and green are set as primary and secondary colors. Blue in this case will be the color displayed most frequently across the screens, whereas the color green will provide more ways to accent and distingush the product.
+
+```jsx
+import { createMuiTheme } from "@material-ui/core/styles";
+import blue from "@material-ui/core/colors/blue";
+import green from "@material-ui/core/colors/green";
+
+const theme = createMuiTheme({
+  palette: {
+    primary: blue,
+    secondary: green,
+  },
+});
+```
+
+## How to change spacings
+
+Spacing is set and configured to create a consistent spacing between the different elements.
+
+Material UI uses a recommended 8px scaling factor by default.
+
+```jsx
+const theme = createMuiTheme();
+
+theme.spacing(2); // = 8 * 2
+```
+
+Custom spacing can also be used and set accordingly:
+
+```jsx
+const theme = createMuiTheme({
+  spacing: 4,
+});
+
+theme.spacing(2); // = 4 * 2
+```
+
+## How to have multiple themes
+
+Mutiple themes can be generated by creating a new instance of `createMuiTheme({})` .
+
+```jsx
+export const baseTheme = createMuiTheme({
+  palette: {
+    primary: {
+      main: purple[500],
+    },
+    secondary: {
+      main: green[500],
+    },
+  },
+});
+
+export const innerTheme = createMuiTheme({
+  palette: {
+    primary: {
+      main: yellow[500],
+    },
+    secondary: {
+      main: black[500],
+    },
+  },
+});
+```
+
+## Light theme vs Dark theme
+
+Material-UI comes with two types of palettes e.g. Light and Dark. The default palette type is Light, but the theme can be made dark by setting `type: "dark"`. While it's only a single property value change, internally it modifies several palette values.
+
+```jsx
+const darkTheme = createMuiTheme({
+  palette: {
+    type: "dark",
+  },
+});
+```
+
+## Themes based on user role
+
+Themes can be rendered based on different user roles. In the example below, we use the `Permission` component and provide a theme only to users that have the role to create (for instance an admin).
+
+```jsx
+import { Permission, PermissionEnum } from "features/permissions/permissions";
+import theme from "styles/theme";
+
+<Permission requiredPermissions={[PermissionEnum.USERS_CREATE]}>
+  <ThemeProvider theme={theme}>{/* your code here*/}</ThemeProvider>
+</Permission>;
```

Comment:
Do we really need to wrap the whole content with `<Permission />` to just change the `theme` based on users' access?

I feel we'll need to write multiple times like below if referencing the example for changing the theme according to access level.
```jsx
<Permission requiredPermissions={[PermissionEnum.USERS_CREATE]}>
  <ThemeProvider theme={theme}>{/* your code here*/}</ThemeProvider>
</Permission>;
<Permission requiredPermissions={[PermissionEnum.DASHBOARD]}>
  <ThemeProvider theme={otherTheme}>{/* your code here*/}</ThemeProvider>
</Permission>;
```

Reply:
agree, I would probably just have a function that returns the theme based on role, for clarity:
```jsx
<ThemeProvider theme={getThemeBasedOnUserRole(user.role)}>
```